axelor-sale/src/main/java/com/axelor/apps/sale/service/SaleOrderLineSaleRepository.java

```diff
@@ -34,8 +35,9 @@ public Map<String, Object> populate(Map<String, Object> json, Map<String, Object
         Beans.get(AppBaseService.class).getNbDecimalDigitForUnitPrice());
 
     if (context.get("_model") != null
-        && context.get("_model").toString().contains("SaleOrder")
-        && context.get("id") != null) {
+        && (context.get("_model").toString().contains("SaleOrder")
+            || context.get("_model").toString().contains("SaleOrderLine"))
```

Comment:
```suggestion
        && (context.get("_model").equals(SaleOrder.class.getName())
            || context.get("_model").equals(SaleOrderLine.class.getName()))
```
So class names are not hard-coded